docs/helpers.rst

```diff
@@ -191,6 +191,26 @@ transaction support.  This is only required for fixtures which need
 database access themselves.  A test function would normally use the
 :py:func:`~pytest.mark.django_db` mark to signal it needs the database.
 
+``shared_db_wrapper``
+~~~~~~~~~~~~~~~~~~~~~
+
+This fixture can be used to create long-lived state in the database.
+It's meant to be used from fixtures with scope bigger than ``function``.
+It provides a context manager that will create a new database savepoint for you,
+and will take care to revert it when your fixture gets cleaned up.
+
+At the moment it does not work with ``transactional_db``,
+as the fixture itself depends on transactions.
+It also needs Django >= 1.6, as it's based on the ``atomic`` API.
+
+Example usage::
+
+  @pytest.fixture(scope='module')
+  def some_users(request, shared_db_wrapper):
+      with shared_db_wrapper(request):
```

Comment:
We need `request` to register a finalizer that will run when the shared fixture (and not just the wrapper) is getting torn down.

Would it be desirable to put all that boilerplate in a decorator that would be used instead of (or maybe before?) pytest.fixture? Something like:
```
@shared_db_fixture(scope='module')
def some_users():
    User.objects.create(username='foo')